pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -0,0 +1,361 @@
+	conditionType := build.Status.Phase
+
+	switch build.Status.Phase {
+	case buildv1.BuildPhaseComplete, buildv1.BuildPhaseCancelled, buildv1.BuildPhaseError, buildv1.BuildPhaseFailed:
+		status = metav1.ConditionTrue
+	}
+
+	return metav1.Condition{
+		Type:               string(conditionType),
+		Status:             status,
+		LastTransitionTime: metav1.Time{Time: time.Now()},
+		Reason: func() string {
+			if string(build.Status.Reason) == "" {
+				return noReasonCondition
+			}
+			return string(build.Status.Reason)
+		}(),
+		Message: fmt.Sprintf("%s%s",
+			build.Name,
+			func() string {
+				if build.Status.Message != "" {
+					return fmt.Sprintf(" => %s", build.Status.Message)
+				}
+				return ""
+			}(),
+		),
+	}
+}
+
+func checkAllBuildsSuccessful(builds map[string]*buildv1.Build) bool {
+	for _, build := range builds {
+		if build.Status.Phase != buildv1.BuildPhaseComplete {
+			return false
+		}
+	}
+	return true
+}
+
+func checkAllBuildsFinished(builds map[string]*buildv1.Build) bool {
+	for _, build := range builds {
+		if build.Status.Phase != buildv1.BuildPhaseComplete &&
+			build.Status.Phase != buildv1.BuildPhaseFailed &&
+			build.Status.Phase != buildv1.BuildPhaseCancelled &&
+			build.Status.Phase != buildv1.BuildPhaseError {
+			return false
+		}
+	}
+	return true
+}
+
+// pushOutputImageWithManifest constructs a manifest-tool command to create and push a new image with all images that we built
+// in the manifest list based on their architecture.
+//
+// Example command:
+// /usr/bin/manifest-tool push from-args \
+// --platforms linux/amd64 --template registry.multi-build01.arm-build.devcluster.openshift.com/ci/managed-clonerefs:latest-amd64 \
+// --platforms linux/arm64 --template registry.multi-build01.arm-build.devcluster.openshift.com/ci/managed-clonerefs:latest-arm64 \
+// --target registry.multi-build01.arm-build.devcluster.openshift.com/ci/managed-clonerefs:latest
+func pushOutputImageWithManifest(builds map[string]*buildv1.Build, targetImageRef string) error {
+	args := []string{"push", "from-args"}
+	for _, build := range builds {
+		args = append(args, []string{
+			"--platforms",
+			fmt.Sprintf("linux/%s", build.Labels[v1.MultiArchBuildConfigArchLabel]),
+			"--template",
+			fmt.Sprintf("%s/%s/%s", registryURL, build.Spec.Output.To.Namespace, build.Spec.Output.To.Name),
+		}...)
+	}
+
+	args = append(args, []string{
+		"--target",
+		fmt.Sprintf("%s/%s", registryURL, targetImageRef),
+	}...)
+
+	cmd := exec.Command("manifest-tool", args...)
+
+	cmdOutput := &bytes.Buffer{}
+	cmdError := &bytes.Buffer{}
+	cmd.Stdout = cmdOutput
+	cmd.Stderr = cmdError
+
+	logrus.Infof("Running command: %s", cmd.String())
+	err := cmd.Run()
+	if err != nil {
+		logrus.WithError(err).WithField("output", cmdOutput.String()).WithField("error_output", cmdError.String()).Info("manifest-tool command failed...")
+		return err
+	}
+
+	logrus.WithField("output", cmdOutput.String()).Info("manifest-tool command succeeded...")
```

Comment:
```suggestion
	logrus.WithField("output", cmdOutput.String()).Info("manifest-tool command succeeded")
```
... suggests that there is something more to it